### PR TITLE
ISPN-10373 Store/Loader Non blocking SPI batch method

### PR DESCRIPTION
**Pull request by wburns**
* Rework batch to allow for atomic write and remove

https://issues.redhat.com/browse/ISPN-10373

The RocksDBStore implementation can be found here https://github.com/infinispan/infinispan/pull/8405/files#diff-f32efbe49859e690ae6743ec1daf7608R210

### CODE REVIEW

**Comment by wburns** [comment]
Need to add some documentation around the new methods still.

**Comment by wburns** [comment]
Added docs.

**Comment by wburns** [comment]
Rebased on latest changes.

File: core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java [hunk 1]
```diff
@@ -885,55 +897,47 @@ private boolean shouldWrite(StoreStatus storeStatus, Predicate<? super StoreConf
...
                checkStoreAvailability();
                if (trace) {
-                  log.tracef("Rolling back transaction %s for stores", transaction);
+                  log.tracef("Rolling back transaction %s for stores", txInvocationContext);
                }
                return Flowable.fromIterable(stores)
                      .filter(storeStatus -> shouldPerformTransactionOperation(storeStatus, predicate))
                      // Let the rollback work in parallel across the stores
-                     .flatMapCompletable(storeStatus -> Completable.fromCompletionStage(storeStatus.store.commit(transaction)));
+                     .flatMapCompletable(storeStatus -> Completable.fromCompletionStage(storeStatus.store.commit(txInvocationContext.getTransaction())));
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java:940]: Shouldn't this be `storeStatus.store.rollback`?

It's worrying that this wasn't detected by our testsuite :scream:

Comment (wburns) [reply core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java:940]: :( My guess it the JDBC store just doesn't have great coverage around this case. I can fix this and I will log a JIRA to create a test case for JDBC store.

Comment (wburns) [reply core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java]: Fixed the typo

**Pull request merged**